zerodb/catalog/indexes/field.py

```diff
@@ -1,11 +1,55 @@
 import itertools as it
-
+import BTrees
+from persistent import Persistent
+from ZODB.broken import Broken
 from repoze.catalog.indexes.field import CatalogFieldIndex as _CatalogFieldIndex
+from repoze.catalog.indexes.common import CatalogIndex
+from repoze.catalog import RangeValue
 from zerodb import trees
 from zerodb.catalog.indexes.common import CallableDiscriminatorMixin
 from zerodb.storage import prefetch
 from zerodb.util.iter import ListPrefetch
 
+_marker = ()
+
+threshold = 10
+
+
+def multiunion1(set_type, seqs):
+    result = set_type()
+    for s in seqs:
+        if (not isinstance(s, BTrees.IFBTree.Bucket) and
```

Comment:
Actually, isinstance can accept a tuple of classes (in this case, it will work faster). E.g. in your case would be:
`not isinstance(s, (Set, BTree, TreeSet, tuple))`. But I think, we can even reduce number of checked types.

But I think that we actually can check whether the item is not `persistent.Persistent` or tuple:
`if not isinstance(s, (Persistent, tuple))`.

Or maybe we can even leverage the fact that we can have only integers or long docids (if we have family64):
`if isinstance(s, (int, long))` and calculating unions only of these here. This would be advantageous to not miss other iterable types. But maybe I'm missing something - can we have non-integers here?